NoSuchFileException

```
PluginId: BashSupport
Plugin name: BashSupport
Plugin version: 1.7.4
IDEA build: CL-183.5153.40

null
java.io.UncheckedIOException: java.nio.file.NoSuchFileException: /usr/bin/deja-dup.dpkg-new
	at java.nio.file.FileTreeIterator.fetchNextIfNeeded(FileTreeIterator.java:88)
	at java.nio.file.FileTreeIterator.hasNext(FileTreeIterator.java:104)
	at java.util.Iterator.forEachRemaining(Iterator.java:115)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at com.ansorgit.plugins.bash.editor.codecompletion.BashPathCompletionService$commands$2.invoke(BashPathCompletionService.kt:62)
	at com.ansorgit.plugins.bash.editor.codecompletion.BashPathCompletionService$commands$2.invoke(BashPathCompletionService.kt:35)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at com.ansorgit.plugins.bash.editor.codecompletion.BashPathCompletionService.getCommands(BashPathCompletionService.kt)
	at com.ansorgit.plugins.bash.editor.codecompletion.BashPathCompletionService.findCommands(BashPathCompletionService.kt:99)
	at com.ansorgit.plugins.bash.editor.codecompletion.PreloadPathCompletionActivity.preload(PreloadPathCompletionActivity.kt:14)
	at com.intellij.openapi.application.Preloader.lambda$null$0(Preloader.java:66)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:164)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:582)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:532)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:87)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:151)
	at com.intellij.openapi.application.Preloader.lambda$initComponent$1(Preloader.java:63)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:227)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$100(BoundedTaskExecutor.java:26)
	at com.intellij.util.concurrency.BoundedTaskExecutor$2$1.run(BoundedTaskExecutor.java:200)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:229)
	at com.intellij.util.concurrency.BoundedTaskExecutor$2.run(BoundedTaskExecutor.java:194)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.nio.file.NoSuchFileException: /usr/bin/deja-dup.dpkg-new
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
	at sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55)
	at sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:144)
	at sun.nio.fs.LinuxFileSystemProvider.readAttributes(LinuxFileSystemProvider.java:99)
	at java.nio.file.Files.readAttributes(Files.java:1737)
	at java.nio.file.FileTreeWalker.getAttributes(FileTreeWalker.java:219)
	at java.nio.file.FileTreeWalker.visit(FileTreeWalker.java:276)
	at java.nio.file.FileTreeWalker.next(FileTreeWalker.java:372)
	at java.nio.file.FileTreeIterator.fetchNextIfNeeded(FileTreeIterator.java:84)
	... 30 more



```

--- a/Changelog.md
+++ b/Changelog.md
@@ -1,5 +1,6 @@
 #### 2019-02-08:
  - [#657](https://github.com/BashSupport/BashSupport/issues/657): Exception when the arithmetic operator >> was used
+ - [#671](https://github.com/BashSupport/BashSupport/issues/671): Ignore/log exception while scanning $PATH, e.g. while a system update is active.
 
 #### 2018-12-15:
  - [#645](https://github.com/BashSupport/BashSupport/issues/645): fix NPE in keyword handling

--- a/src/com/ansorgit/plugins/bash/editor/codecompletion/BashPathCompletionService.kt
+++ b/src/com/ansorgit/plugins/bash/editor/codecompletion/BashPathCompletionService.kt
@@ -37,7 +37,7 @@ class BashPathCompletionService() {
         private val LOG = Logger.getInstance("#bash.completion")
 
         @JvmStatic
-        fun getInstance() = ServiceManager.getService(BashPathCompletionService::class.java)
+        fun getInstance() = ServiceManager.getService(BashPathCompletionService::class.java)!!
     }
 
     data class CompletionItem(val filename: String, val path: String)
@@ -81,8 +81,9 @@ class BashPathCompletionService() {
                         }
                     } catch (ex: Exception) {
                         when (ex) {
-                            is InvalidPathException, is IOException, is SecurityException -> LOG.warn("Invalid path detected in \$PATH element $e", ex)
-                            else -> throw ex
+                            is InvalidPathException, is IOException, is SecurityException -> LOG.debug("Invalid path detected in \$PATH element $e", ex)
+                            is FileSystemException -> LOG.debug("Ignoring filesystem exception in \$PATH element $e", ex)
+                            else -> LOG.error("Exception while scanning \$PATH for command names", ex)
                         }
                     }
                 }

